src/plugins/wizard/public/application/utils/state_management/metadata_slice.ts

```diff
@@ -12,12 +12,20 @@ export interface MetadataState {
       // Validity for each section in the editor
       [key: string]: boolean;
     };
+    errMsg: string;
+    hasChange: boolean;
+    isEmpty: boolean;
+    hasUnappliedChanges: boolean;
   };
 }
 
 const initialState: MetadataState = {
   editorState: {
     validity: {},
+    errMsg: '',
```

Comment:
I wouldn't store this on the state. Since the error message depends on where we are displaying the error.